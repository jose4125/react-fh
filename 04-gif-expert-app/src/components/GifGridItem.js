import PropTypes from "prop-types";

export default function GifGridItem({ title, url }) {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title} />
      <h2>{title}</h2>
    </div>
  );
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
