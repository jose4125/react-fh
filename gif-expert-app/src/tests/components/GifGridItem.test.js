import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("GifGridItem tests", () => {
  let wrapper;
  const props = {
    title: "goku image",
    url: "https://localhost/image.jpg",
  };

  beforeEach(() => {
    wrapper = shallow(<GifGridItem {...props} />);
  });

  test("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render title", () => {
    const title = wrapper.find("h2").text();
    expect(title).toBe(props.title);
  });

  test("should render image", () => {
    const src = wrapper.find("img").prop("src");
    expect(src).toBe(props.url);
  });

  test("should image should have alt text", () => {
    const alt = wrapper.find("img").prop("alt");
    expect(alt).toBe(props.title);
  });

  test("should have fadein class", () => {
    const animClass = wrapper.find(".card").prop("className");
    expect(animClass).toContain("animate__fadeIn");
  });
});
