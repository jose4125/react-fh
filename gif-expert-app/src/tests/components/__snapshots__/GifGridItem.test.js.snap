// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem tests should render properly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="goku image"
    src="https://localhost/image.jpg"
  />
  <h2>
    goku image
  </h2>
</div>
`;
